ui/StatusQ/src/StatusQ/Core/StatusIcon.qml

```diff
@@ -13,8 +13,13 @@ Image {
     fillMode: Image.PreserveAspectFit
 
     onIconChanged: {
-        if (icon !== "") {
-            source = "../../assets/img/icons/" + icon + ".svg";
+        if(icon.startsWith("data:image/") || icon.startsWith("https://")) {
+            //raw image data
+            source = icon
+            objectName = "custom-icon"    
+        }
+        else if (icon !== "") {
+            source = "../../assets/img/icons/" + icon;
```

Comment:
Isn't `+ ".svg"` missing?

Reply:
Qt can detect the extension if needed
https://doc.qt.io/qt-5/qml-qtquick-image.html#automatic-detection-of-file-extension

Reply:
Nice, didn't know that

Reply:
@alexjba please put the extension back :) Even though Qt can indeed detect the image type from the extension, it will not set the correct `sourceSize` for SVGs, resulting in blurry images on highres displays

Reply:
Ohh, didn't knew of this issue 😶‍🌫️